### DISCUSSION TITLE
Is there any problem with the Rearrange I wrote?

### DISCUSSION THREAD

**Discussion by chaoji90** [arogozhnikov/einops#141] [Q&A] [answered]
I want to reshape tensor from (batch_size, 256, 64) to (batch_size, 16, 32, 32), I wrote the following:
Rearrange('b (c1 c2 c3) (h1 w1))->b c1 (c2 h1) (c3 w1)',c1=16, c2=4, c3=4, h1=8,w1=8)
and the error reported "Brackets are not balanced"
I'm not sure what is the problem here and what should be written is appropriate.

**Comment by arogozhnikov** [comment, marked as answer]
Replaced redundant ) with ] from your example, you need to remove it:
```
Rearrange('b (c1 c2 c3) (h1 w1)]->b c1 (c2 h1) (c3 w1)',c1=16, c2=4, c3=4, h1=8,w1=8)
```

**Reply by chaoji90** [reply]
Thanks a lot
